app/Umbraco/Umbraco.Archetype/PropertyEditors/ArcheTypePropertyEditor.cs

```diff
@@ -216,7 +215,7 @@ public override object ConvertEditorToDb(ContentPropertyData editorValue, object
 							var propData = new ContentPropertyData(propDef.Value, preValues, additionalData);
 							var propEditor = PropertyEditorResolver.Current.GetByAlias(dtd.PropertyEditorAlias);
 							// make sure to send the current property value (if any) to the PE ValueEditor
-							propDef.Value = propEditor.ValueEditor.ConvertEditorToDb(propData, currentProperty != null ? currentProperty.Value : null);
+							propDef.Value = propEditor.ValueEditor.ConvertEditorToDb(propData, currentProperty != null ? currentProperty.Value.ToString() : null);
```

Comment:
You'll probably want to add null checks to currentProperty.Value before executing a ToString() on it otherwise you can end up with null reference exceptions.